src/main/scala/dpla/ingestion3/harvesters/api/MdlHarvester.scala

```diff
@@ -95,18 +102,40 @@ class MdlHarvester(spark: SparkSession,
   }
 
   /**
-    * Builds MDL query URL from parameters
+    *
+    * First MDL URL from parameters
+    *
+    * Base URL https://lib-metl-prd-01.oit.umn.edu/api/v2/records
     *
     * @param queryParams URL parameters
     * @return URI
     */
-  def buildUrl(queryParams: Map[String, String]): URL = new URIBuilder()
-    .setScheme("http")
-    .setHost("hub-client.lib.umn.edu")
-    .setPath("/api/v1/records")
-    .setParameter("q", queryParams.getOrElse("query", "*:*"))
-    .setParameter("start", queryParams.getOrElse("start", "0"))
-    .setParameter("rows", queryParams.getOrElse("rows", "10"))
-    .build()
-    .toURL
+  def getFirstUrl(queryParams: Map[String, String]): String = {
+    val url = new URIBuilder()
+      .setScheme("https")
+      .setHost("lib-metl-prd-01.oit.umn.edu")
+      .setPath("/api/v2/records")
+      .setParameter("page[size]", queryParams.getOrElse("rows", defaultRows))
+
+
+    /**
+      * This is a real mangling of shit
+      */
+    val params = queryParams
+      .getOrElse("query", defaultQuery)
+      .split(", ")
+      .map( query => {
+        val query_tuple = query.split("=")
+        new Parameters(query_tuple(0), query_tuple(1))
+      })
+    // Add query parameters to URL
+    params.foreach(p => url.addParameter(p.getName, p.getValue))
+
+    url.build().toURL.toString
+  }
 }
+
+class Parameters(name: String, value: String) {
```

Comment:
case class? then you don't need the accessor methods.

Reply:
done